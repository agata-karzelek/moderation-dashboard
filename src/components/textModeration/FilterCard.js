import React from 'react';
import { Grid, Box, Typography } from '@material-ui/core';
import { useStyles } from '../../style/textModeration.js';

const FilterCard = ({ state, handleClick }) => {
  const classes = useStyles();
  return (
    <Grid container justify="flex-start" spacing={3}>
      <Grid item>
        <Box
          className={
            state.wordListProfanity === 'true'
              ? classes.methodSelected
              : classes.methodNotSelected
          }
          onClick={handleClick('wordListMethod')}
        >
          <Grid container justify="center">
            <Box p={1}>
              <img
                alt=""
                src={
                  state.wordListProfanity === 'true'
                    ? '/images/selected-word-list.svg'
                    : '/images/word-list-not-selected.svg'
                }
              />
            </Box>
          </Grid>
          <Typography variant="body1" align="center">
            Word List
          </Typography>
        </Box>
      </Grid>
      <Grid item>
        <Box
          className={
            state.automaticProfanity === 'true'
              ? classes.methodSelected
              : classes.methodNotSelected
          }
          onClick={handleClick('automaticMethod')}
        >
          <Grid container justify="center">
            <Box p={1}>
              <img
                alt=""
                src={
                  state.automaticProfanity === 'true'
                    ? '/images/automatic-profanity-selected.svg'
                    : '/images/automatic-detection.svg'
                }
              />
            </Box>
          </Grid>
          <Typography testid="autoDetection" variant="body1" align="center">
            Automatic Detection
          </Typography>
        </Box>
      </Grid>
    </Grid>
  );
};

export default FilterCard;
