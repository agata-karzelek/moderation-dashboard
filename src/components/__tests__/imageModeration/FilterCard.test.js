import React from "react";
import { shallow } from "enzyme";
import FilterCard from "../../imageModeration/FilterCard";

const mockProps = {
  state: {
    initialLoading: false,
  },
};

describe("Test Case for Filter Card", () => {
  test("Snapshot", () => {
    const wrapper = shallow(<FilterCard state={mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("check text", () => {
    const wrapper = shallow(<FilterCard state={mockProps} />);
    expect(wrapper.find('[testid="autoDetection"]').text()).toBe("Automatic Detection");
  });
});
