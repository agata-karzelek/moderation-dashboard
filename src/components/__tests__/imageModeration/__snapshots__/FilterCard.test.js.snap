// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case for Filter Card Snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  justify="flex-start"
  spacing={3}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(MuiBox)
      className="makeStyles-methodSelected-10"
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="center"
      >
        <Styled(MuiBox)
          p={1}
        >
          <img
            alt=""
            src="/images/automatic-profanity-selected.svg"
          />
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Typography))
        align="center"
        testid="autoDetection"
        variant="body1"
      >
        Automatic Detection
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
